prov/shm/src/smr_util.c

```diff
@@ -544,40 +587,22 @@ int smr_map_add(const struct fi_provider *prov, struct smr_map *map,
 	return FI_SUCCESS;
 }
 
-void smr_map_del(struct smr_map *map, int64_t id)
+int smr_map_del(struct ofi_rbmap *rbmap, struct ofi_rbnode *node,
+		 void *context)
 {
-	struct dlist_entry *entry;
+	struct smr_map *map = container_of(rbmap, struct smr_map, rbmap);
+	int64_t id = (uintptr_t) node->data;
 
 	assert(id >= 0 && id < SMR_MAX_PEERS);
-
-	pthread_mutex_lock(&ep_list_lock);
-	entry = dlist_find_first_match(&ep_name_list, smr_match_name,
-				       smr_no_prefix(map->peers[id].peer.name));
-	pthread_mutex_unlock(&ep_list_lock);
-
+	smr_unmap_region(&smr_prov, map, id);
 	ofi_spin_lock(&map->lock);
-	(void) ofi_rbmap_find_delete(&map->rbmap,
-				     (void *) map->peers[id].peer.name);
-
+	ofi_rbmap_delete(rbmap, node);
```

Comment:
My hesitation with this is it requires the cleanup path to have a lot of unnecessary rebalances. it's not a big deal but I think it would make more sense to have the iterative function not free the node (and have the rbmap_cleanup right after handle all the frees). that one can be `smr_map_unmap` or something and then have the `smr_map_del` call in av_insert call the iterative one (map_unmap) and then rbmap_delete which will selectively delete it and rebalance if necessary